src/entities/Proposal/jobs.ts

```diff
@@ -162,17 +157,22 @@ export async function finishProposal(context: JobContext) {
   if (rejectedProposals.length > 0) {
     context.log(`Rejecting ${rejectedProposals.length} proposals...`)
     await ProposalModel.finishProposal(
-      rejectedProposals.map(({ proposal }) => proposal.id),
+      rejectedProposals.map(({ id }) => id),
       ProposalStatus.Rejected
     )
-    rejectedProposals.map(({ proposal }) =>
-      DiscordService.finishProposal(proposal.id, proposal.title, ProposalOutcome.REJECTED)
-    )
   }
 
   const proposals: ProposalWithWinnerChoice[] = [...finishedProposals, ...acceptedProposals, ...rejectedProposals]
   context.log(`Updating ${proposals.length} proposals in discourse... \n\n`)
-  for (const { proposal } of proposals) {
-    commentProposalUpdateInDiscourse(proposal.id)
+  for (const { id, title, winnerChoice, outcomeStatus } of proposals) {
+    commentProposalUpdateInDiscourse(id)
+    if (outcomeStatus) {
+      DiscordService.finishProposal(
```

Comment:
I think we might be missing an `await`

We could wrap this with `inBackground()`, as in `commentProposalUpdateInDiscourse`